File: JAVA/Multilevel.java
class Base
{
    public int i;
    public int j;

    public Base()   //Default
    {
        System.out.println("Inside Base Constructor");
    }
    
    public void Fun()
    {
        System.out.println("Inside Base Fun()");
    }
    public void Gun()
    {
        System.out.println("Inside Base Gun()");
    }

}
class Derived extends Base
{
    public int x;
    public int y;

    public Derived()
    {
        System.out.println("Inside Derived Constructor");
    }

    public void Sun()
    {
        System.out.println("Inside Derived Sun()");
    }
}
class DerivedX extends Derived
{
    public int A;

    public DerivedX()
    {
        System.out.println("Inside DerivedX Constructor");
    }
    public void Run()
    {
        System.out.println("Inside Derived Run()");
    }
}





class Multilevel
{
    public static void main(String A[])
    {
            DerivedX dobj = new DerivedX();

            dobj.Fun();
            dobj.Gun();
            dobj.Sun();
            dobj.Run();
    }
}

File: CPP/Multilevel.cpp
        #include <iostream>
using namespace std;

class Base
{
    public:

        int i, j;

        Base()
        {
        cout<<"Inside Base Constructor \n";
        }

        void Fun()
        {
            cout<<"Inside Base Fun() \n";
        }
        void Gun()
        {
            cout<<"Inside Base Gun() \n";
        }
        ~Base()
        {
        cout<<"Inside Base Destructor \n";
        }
};
class Derived : public Base
{
    public:
       

        int x, y;

        void  Sun()
        {
            cout<<"Inside Base Sun() \n";
        }

     
        Derived()
        {
            cout<<"Inside Derived Constructor \n";
        }
        ~Derived()
        {
            cout<<"Inside Derived Destructor \n";
        }
     
};
class DerivedX : public Derived
{
    public:
        int A;

        DerivedX()
        {
            cout<<"Inside DerivedX Constructor \n";
        }
        ~DerivedX()
        {
            cout<<"Inside DerivedX Destructor \n";
        }
        void Run()
        {
            cout<<"Inside DerivedX Run() \n";
        }

};

int main()
{   
        DerivedX dobj;
        
        dobj.Fun();
        dobj.Gun();
        dobj.Sun();
        dobj.Run();


    return 0;
}
